### PR TITLE
fix: cleanup after `pages dev` tests

### PR DESCRIPTION
**Pull request by threepointone**
We weren't killing the process started by wrangler whenever its parent was killed. This fix is to listen on SIGINT/SIGTERM and kill that process. I also did some minor configuration cleanups.

Fixes https://github.com/cloudflare/wrangler2/issues/397
Fixes https://github.com/cloudflare/wrangler2/issues/618

--- 

(fyi this does NOT cleanup the occasional flake with the remix test where the text doesn't render)

### CODE REVIEW

File: packages/example-remix-pages-app/tests/index.test.ts [hunk 1]
```diff
@@ -26,23 +26,32 @@ describe("Remix", () => {
...
     });
-    wranglerProcess.stderr.on("data", (chunk) => {
+    wranglerProcess.stderr?.on("data", (chunk) => {
       console.log(chunk.toString());
     });
   });
 
-  afterAll(() => {
-    wranglerProcess.kill();
+  afterAll(async () => {
```
Comment (threepointone) [inline packages/example-remix-pages-app/tests/index.test.ts:44]: could probably clean this up by using execa here, but I'll leave that for some other time/person

**Comment by threepointone** [comment]
oh lol this hangs on windows and ubuntu huh

**Comment by threepointone** [comment]
Aight, looks like forceExit is still necessary for windows/ubuntu.

File: packages/wrangler/src/pages.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ const EXIT = (message?: string, code?: number) => {
   if (message) console.log(message);
   if (code) process.exitCode = code;
   EXIT_CALLBACKS.forEach((callback) => callback());
+  RUNNING_BUILDERS.forEach((builder) => builder.stop?.());
```
Comment (threepointone) [inline packages/wrangler/src/pages.tsx:30]: This cleans up hanging esbuild processes

**Review by GregBrimble — APPROVED** [review]
Sorry, thought we'd got it yesterday! Let's see if this does it now.

**Comment by threepointone** [comment]
🤞

**Pull request merged**